bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,356 @@
+                case TEMPERATURE:
+                    return new QuantityType<>(goeResponse.getTemperature(), SIUnits.CELSIUS);
+                case SESSION_CHARGE_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.getSessionChargeConsumption() / 360000d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case SESSION_CHARGE_CONSUMPTION_LIMIT:
+                    return new QuantityType<>((Double) (goeResponse.getSessionChargeConsumptionLimit() / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case TOTAL_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.getTotalChargeConsumption() / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case FIRMWARE:
+                    return new StringType(goeResponse.getFirmware());
+                case VOLTAGE_L1:
+                    return new QuantityType<>(goeResponse.getEnergy()[0], SmartHomeUnits.VOLT);
+                case VOLTAGE_L2:
+                    return new QuantityType<>(goeResponse.getEnergy()[1], SmartHomeUnits.VOLT);
+                case VOLTAGE_L3:
+                    return new QuantityType<>(goeResponse.getEnergy()[2], SmartHomeUnits.VOLT);
+                case CURRENT_L1:
+                    return new QuantityType<>((Double) (goeResponse.getEnergy()[4] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L2:
+                    return new QuantityType<>((Double) (goeResponse.getEnergy()[5] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L3:
+                    return new QuantityType<>((Double) (goeResponse.getEnergy()[6] / 10d), SmartHomeUnits.AMPERE);
+                case POWER_L1:
+                    return new QuantityType<>((Double) (goeResponse.getEnergy()[7] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L2:
+                    return new QuantityType<>((Double) (goeResponse.getEnergy()[8] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L3:
+                    return new QuantityType<>((Double) (goeResponse.getEnergy()[9] / 10d),
+                            SmartHomeUnits.WATT);
+                default:
+                    return UnDefType.NULL;
+            }
+        }
+        return UnDefType.NULL;
+    }
+
+    /**
+     * Update the channel from the last data
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    protected void updateChannel(String channelId) {
+        if (!isLinked(channelId)) {
+            return;
+        }
+
+        State state = getValue(channelId);
+        updateState(channelId, state);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId());
+            return;
```

Comment:
I might be wrong, but I think the purpose of this command is to retrieve current value for given channel directly from the device, so I would call `refresh()` here.
@J-N-K  WDYT?

Reply:
good catch, this also works good together with the review remark a few comments above to change 

```
getThing().getChannels().forEach(channel -> updateChannel(channel.getUID().getId()));
```
to
```
getThing().getChannels().map(channel -> channel.getUID().getId()).forEach(id -> updateState(id, getValue(id)));
```

How often is this refresh command triggered? in the end i already refresh like every 5 seconds or so, if there is an additional refresh inbetween i dont think that gives you more/better data.